src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -391,24 +360,101 @@ private void LowerBasicSwitch(DecisionTree.ByValue byValue)
                 }
```

Comment:
This code does not actually assume or depend on the fact that it is *not* boolean, so such an assertion would be out of place.